include/net/net_context.h

```diff
@@ -27,6 +27,12 @@
 #include <net/net_if.h>
 #include <net/net_stats.h>
 
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+#if defined(CONFIG_MBEDTLS)
+#include <mbedtls/ssl.h>
```

Comment:
This adds dependency of the application-level protocol(s), TLS/DTLS, to a networking context, and that doesn't make sense technically - That's layering violation. What next we will make the networking context depend on? HTTP? MQTT? LWM2M?  

(This goes beyond discussing paradigmatic/API design issues in https://github.com/zephyrproject-rtos/zephyr/pull/5985#issuecomment-383177190 to discussing technical issues with such approach. Which, as was mentioned previously, are many too).